crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -1003,6 +997,14 @@ pub fn verify_nsec(query: &Query, soa_name: &Name, nsecs: &[&Record]) -> Proof {
     }
 }
 
+/// Returns the current system time as Unix timestamp in seconds.
+fn current_time() -> u32 {
+    SystemTime::now()
```

Comment:
does not need to be done in this PR but we need tests that artificially set the time to more than `u32::MAX` seconds elapsed from UNIX_EPOCH. see https://github.com/hickory-dns/hickory-dns/issues/2312